R/teal_data_utils.R

```diff
@@ -0,0 +1,69 @@
+#' `teal_data` utils
+#'
+#' In `teal` we need to recreate the `teal_data` object due to two operations:integer.max
+#' - we need to append filter-data code and objects which have been evaluated in `FilteredData` and
+#' we want to avoid double-evaluation.
+#' - we need to subset `teal_data` to datanames used by the module, to shorten obtainable R-code
+#'
+#' Due to above recreation of `teal_data` object can't be done simply by using public
+#' `teal.code` and `teal.data` methods.
+#'
+#' @param data (`teal_data`)
+#' @param code (`character`) code to append to `data@code`
+#' @param objects (`list`) objects to append to `data@env`
+#' @param datanames (`character`) names of the datasets
+#' @return modified `teal_data`
+#' @name teal_data_utilities
+NULL
+
+#' @rdname teal_data_utilities
+.append_evaluated_code <- function(data, code) {
+  checkmate::assert_class(data, "teal_data")
+  data@code <- c(data@code, code)
+  data@id <- c(data@id, sample.int(.Machine$integer.max, size = length(code)))
+  data@messages <- c(data@messages, rep("", length(code)))
+  data@warnings <- c(data@warnings, rep("", length(code)))
+  methods::validObject(data)
+  data
+}
+
+#' @rdname teal_data_utilities
+.append_modified_data <- function(data, objects) {
+  checkmate::assert_class(data, "teal_data")
+  checkmate::assert_class(objects, "list")
+  new_env <- list2env(objects, parent = .GlobalEnv)
+  rlang::env_coalesce(new_env, data@env)
```

Comment:
Does this copies bindings from `data@env` to `new_env` or it does `c()` operations that does `union`?
Trying to figure out the purpose of this line!

Reply:
Ah, thanks! Gotcha, all clear. Really helpful that I asked for this line.